pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -84,8 +84,8 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 	// compute the proxy's locality. See if we have a CDS cache for that locality.
 	// If not, compute one.
 	locality := proxy.Locality
-	if locality != nil {
-		applyLocalityLBSetting(locality, clusters, env.Mesh.LocalityLbSetting, false)
+	if locality != nil && pilot.EnableLocalityLoadBalancing() {
```

Comment:
what is this feature flag? Lets not add a pilot level env variable to globally enable/disable this. Its bad UX. Please add this to the global mesh config, under Pilot or in the global settings. Users need to be able to see this in helm chart and enable/disable

Reply:
This flag is not a new one, added in release 1.1 by @liamawhite , here i just find we did not protect the setting by feature flag.

Reply:
If we're enabling by default, it should be controllable by global settings.

Reply:
I donot think they are related. Are there any doc saying that?

Reply:
As @rshriram said. A hidden env var to turn it off is not a good UX.

Reply:
@hzxuzhonghu what I am saying is that you should remove this pilot.Enable.. and instead have something like `env.LocalityAwareRouting == true` condition.. and this setting in env must come from mesh/v1alpha1/config.proto . This will then let people enable the feature from the helm chart or any other means of generating that config map.

Reply:
IC, will do that.  I'd like a feature configuration api for pilot, eventually all feature envs goes into it.